lib/src/presentation/scenes/save_habit/save_habit.dart

```diff
@@ -91,43 +92,60 @@ class _SaveHabitState extends State<SaveHabit> {
         onTap: () {
           FocusScope.of(context).unfocus();
         },
-        child: Container(
-          color: XHColors.darkGrey,
-          padding: SizeConfig.saveScreenPadding,
-          child: ListView(
-            children: <Widget>[
-              Text(
-                'What do you want to accomplish?',
-                style: TextStyle(
-                  fontSize: SizeConfig.saveScreenLargeText,
-                  color: Colors.white,
-                ),
-              ),
-              Container(
-                margin: SizeConfig.saveScreenInputMargin,
-                child: TextField(
-                  controller: _titleController,
+        child: WillPopScope(
+          onWillPop: () => Future(() {
+              if (_saveHabitBloc.dataEntered) {
+                ConfirmDialog.show(
+                    context,
+                    'Leave the screen?',
```

Comment:
@phenybutum maybe it would make more sense providing a contextual message? Like 
"Leave without saving?"